File: kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/producer/headers/ProductClient.java
package io.micronaut.configuration.kafka.docs.producer.headers;

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.messaging.annotation.MessageHeader;
// end::imports[]

// tag::clazz[]
@KafkaClient(id="product-client")
@MessageHeader(name = "X-Token", value = "${my.application.token}")
public interface ProductClient {
// end::clazz[]
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/consumer/config/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.configuration.kafka.annotation import KafkaClient, KafkaKey, Topic
from micronaut.context.annotation import Requires
from micronaut.kafka.docs.Product import Product


@Requires(property="spec.name", value="ConfigProductListenerTest")
@KafkaClient("product-client")
class ProductClient(ABC):

    @Topic("awesome-products")
    @abstractmethod
    def send(self, brand: Annotated[str, KafkaKey], product: Product) -> None:
        ...
